tsconfig.json

```diff
@@ -12,7 +12,7 @@
     "sourceMap": true,
     "skipLibCheck": true,
 
-    "lib": ["es2020", "dom", "dom.iterable"],
+    "lib": ["es2022", "dom", "dom.iterable"],
```

Comment:
This change is to use `Object.hasOwn()` static method.
If this change should be avoided, I will use `hasOwnProperty()` instance method instead.

Reply:
I don't think hasOwn/hasOwnProperty is even needed there. That logic should probably be

```ts
.every((v) => value[v] === null || typeof value[v] === 'number') // === 'string' in isHms
```

Reply:
At the point of `value[v]`, type of value is `object` and TypeScript throws an type error:

> Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
> No index signature with a parameter of type 'string' was found on type '{}'.

This is why I thought it is needed to use either of `hasOwn/hasOwnProperty`.
The reason I used `hasOwn` over `hasOwnProperty` is just it's free from prototype pollution.

Is there any way to resolve this type error, or do you think it's reasonable enough to use `as any`?

Reply:
Casting value as any is fine IG.

Reply:
You can make types of reduce stricter too:

```ts
export function hasProp<T extends string>(value: Record<string, unknown>, prop: T): value is { [key in T]: unknown } {
  return Object.hasOwn(value, prop)
}

export function isYmd(value: unknown, required: YmdType[] = ['y', 'm', 'd']): value is Ymd {
  if (!isObject(value)) {
    return false
  }

  return required
    .reduce<(keyof Ymd)[]>((keys, type) => {
      keys.push(YmdMap[type])
      return keys
    }, [])
    .every((v) => hasProp(value, v) && isNumber(value[v]))
}

export function isHms(value: unknown, required: HmsType[] = ['h', 'm', 's']): value is Hms {
  if (!isObject(value)) {
    return false
  }

  return required
    .reduce<(keyof Hms)[]>((keys, type) => {
      keys.push(HmsMap[type])
      return keys
    }, [])
    .every((v) => hasProp(value, v) && isString(value[v]))
}
```

Reply:
> BTW, if you use isObject it won't throw any TS errors:

Oh yes, it has type predicate.
Thank you, fixed: 4a4eddb